packages/testing/src/mocking-activity-environment.ts

```diff
@@ -30,7 +30,7 @@ export interface MockActivityEnvironmentOptions {
  * will immediately be in a cancelled state.
  */
 export class MockActivityEnvironment extends events.EventEmitter {
-  public cancel: (reason?: any) => void = () => undefined;
+  public cancel: (reason?: any, details?: any) => void = () => undefined;
```

Comment:
What would a `MockActivityEnvironment` user be expected to provide as input for the `details` argument? The signature says `any`, and the `ActivityCancellationDetails` class has a private constructor, so it looks like it is simply impossible for users to provide anything useful here.

One solution would be to have `ActivityCancellationDetails` be just an interface, rather than a class, but then we'd have to make all properties optional, as adding more fields in the future would be a backward incompatible change.

Can you please look at what we do with this in other SDKs, and try to figure out a proper API for this?

And btw, `reason` being `any` also feels wrong.